Add TTL to CREATE and ALTER TABLE (#5063)
ydb-platform/ydb: ydb/docs/en/core/yql/reference/yql-core/syntax/_includes/create_table.md

For a list of valid parameter names and values, see the [{{ backend_name }} table description]({{ concept_table }}).

> ### Examples of Creating a String Table with Additional Parameters

Reviewer: ```suggestion
### Examples of creating a table with additional parameters
```
This is a common way to write headers, but we're not currently using it in other places.

> This code will create a table with enabled automatic partitioning by partition size and the preferred size of each partition is 512 MB:

Reviewer: ```suggestion
This code will create a table with enabled automatic partitioning by size and the preferred partition size of 512 MB:
```

);
```

> Creating a table with enabled [TTL](../../../../concepts/ttl.md) for automatically removing outdated records from the table. A record is deleted when the value of one of the columns (configurable by the user) of type: `Date`, `Datetime`, or `Timestamp` becomes greater than the current time (Epoch time). The lifespan of records can be extended (globally for the entire table) by the value specified in the `Interval` function:

Reviewer: ```suggestion
Creating a table with enabled [TTL](../../../../concepts/ttl.md) to asynchronously remove outdated records from the table automatically. {{ ydb-short-name }} periodically checks if each record needs to be deleted by checking if the value of a specified column plus the provided interval is greater than the current time:
```

```sql
CREATE TABLE my_table (
    id Uint64,
    title Utf8,
    expire_at Timestamp
    PRIMARY KEY (id)
)
WITH (

>     TTL = Interval("PT0S") ON expire_at

Reviewer: This is a corner case example. Let's add a more typical one first. For example, there's a column with an event time (timestamp at the moment of insert), and we want to discard events older than one week.

>     TTL = Interval("PT0S") ON expire_at

Reviewer: According to https://ydb.tech/docs/en/concepts/ttl, there's also an option to use an integer column and specify in which time unit to interpret values:

> The value in the TTL column with a numeric type (Uint32, Uint64, or DyNumber) is interpreted as a [Unix time](https://en.wikipedia.org/wiki/Unix_time) value. The following units are supported (set in the TTL settings):
> 
> - Seconds.
> - Milliseconds.
> - Microseconds.
> - Nanoseconds.

It would be helpful to cover this case, too.

);
```

> In the provided example, a record will be deleted as soon as the value in the `expire_at` column matches the Epoch time, since the Interval function specifies the value "PT0S" is 0 seconds.

Reviewer: ```suggestion
In the provided example, the specified interval is `"PT0S"`, which according to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601), means zero seconds. Thus, records will be deleted when the value in the `expire_at` column exceeds the current Epoch time.
```

{% note info %}

> In the table, there can be only one TTL column. If in the TTL column has a `NULL` value, that row will never be deleted.

Reviewer: ```suggestion
There can be only one TTL column in a table. If the TTL column has a `NULL` value, that row will never be deleted.
```

> In the table, there can be only one TTL column. If in the TTL column has a `NULL` value, that row will never be deleted.

Reviewer: In this patch, it looks like "rows" and "records" are used as synonyms. Let's stick to one of them to minimize the risk of confusion.
